File: Plus One.py
class Solution:
    # @param digits, a list of integer digits
    # @return a list of integer digits
    def plusOne(self, digits):
        if (len(digits) == 0):
            return digits
        index = len(digits) - 1
        while (index >= 0):
            digits[index] = digits[index] + 1;
            if (digits[index] < 10):
                return digits;
            # carry to next digit
            digits[index] = 0;
            index = index - 1;
        ret = [1]
        ret.extend(digits)
        return ret

File: Plus One.java
public class Solution {
    public int[] plusOne(int[] digits) {
        if (digits.length == 0)
            return digits;
        int i = digits.length - 1;
        while (true) {
            digits[i]++;
            if (digits[i] < 10)
                return digits;
            digits[i] = 0;
            i--;
            if (i < 0) {
                int[] ret = new int[digits.length + 1];
                ret[0] = 1;
                System.arraycopy(digits, 0, ret, 1, digits.length);
                return ret;
            }
        }
    }
}